src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/finance/TransactionsService.kt

```diff
@@ -0,0 +1,180 @@
+        reference = it.transactionReferenceNumber,
+        clientReference = it.clientUniqueRef,
+        subAccountType = it.subAccountType.name,
+        description = it.entryDescription ?: "",
+        createdAt = it.createDatetime,
+        createdBy = it.createUsername,
+        createdByDisplayName = getDisplayName(it.createStaffUserAccount),
+        lastModifiedAt = it.modifyDatetime,
+        lastModifiedBy = it.modifyUserId,
+        lastModifiedByDisplayName = getDisplayName(it.modifyStaffUserAccount),
+        generalLedgerTransactions = it.generalLedgerTransactions.map(::mapGL),
+      )
+    }
+
+  fun getGeneralLedgerTransactions(
+    transactionId: Long,
+  ): List<GeneralLedgerTransactionDto> = generalLedgerTransactionRepository
+    .findByTransactionId(transactionId)
+    .map(::mapGL)
+}
+
+private fun mapGL(gl: GeneralLedgerTransaction): GeneralLedgerTransactionDto = GeneralLedgerTransactionDto(
+  transactionId = gl.transactionId,
+  transactionEntrySequence = gl.transactionEntrySequence,
+  generalLedgerEntrySequence = gl.generalLedgerEntrySequence,
+  caseloadId = gl.caseloadId,
+  amount = gl.entryAmount,
+  type = gl.transactionType.type,
+  accountCode = gl.accountCode.accountCode,
+  description = gl.entryDescription ?: "",
+  transactionTimestamp = gl.entryDate.atTime(gl.entryTime),
+  reference = gl.transactionReferenceNumber,
+  createdAt = gl.createDatetime,
+  createdBy = gl.createUsername,
+  createdByDisplayName = getDisplayName(gl.createStaffUserAccount),
+  lastModifiedAt = gl.modifyDatetime,
+  lastModifiedBy = gl.modifyUserId,
+  lastModifiedByDisplayName = getDisplayName(gl.modifyStaffUserAccount),
+)
+
+private fun getDisplayName(staffUserAccount: StaffUserAccount?): String = staffUserAccount
+  ?.staff?.run { "$firstName $lastName" } ?: "Unknown"
+
+@Schema(description = "The data held in NOMIS about a financial transaction")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class OffenderTransactionDto(
+  @Schema(description = "The transaction id", example = "12345678")
+  val transactionId: Long,
+
+  @Schema(description = "The sequence number", example = "3")
+  val transactionEntrySequence: Int,
+
+  @Schema(description = "The offender id", example = "1234567")
+  val offenderId: Long,
+
+  @Schema(description = "The prisoner number or noms id", example = "A1234AA")
+  val offenderNo: String,
+
+  @Schema(description = "The booking id", example = "12345678")
+  val bookingId: Long?,
+
+  @Schema(description = "The caseload", example = "MDI")
+  val caseloadId: String,
+
+  @Schema(description = "The transaction amount", example = "2.14")
+  val amount: BigDecimal,
+
+  @Schema(description = "The transaction type defined in the TRANSACTION_TYPES table", example = "CANT")
+  val type: String,
+
+  @Schema(description = "Whether credit or debit", allowableValues = ["CR", "DR"])
+  val postingType: String,
+
+  @Schema(description = "A description of the transaction entry", example = "???")
```

Comment:
Probably no example is better than ???

Reply:
whoah lol, i read your comment as "??? is the best possible example" ... sorted